vault/acl.go

```diff
@@ -47,25 +47,83 @@ func NewACL(policies []*Policy) (*ACL, error) {
 			// Check for an existing policy
 			raw, ok := tree.Get(pc.Prefix)
 			if !ok {
-				tree.Insert(pc.Prefix, pc.CapabilitiesBitmap)
+				tree.Insert(pc.Prefix, pc.Permissions)
 				continue
 			}
-			existing := raw.(uint32)
+
+			// these are the ones already in the tree
+			permissions := raw.(*Permissions)
```

Comment:
I think this is a bit confusing naming -- I'd name it something like existingPerms and just get rid of `existing` in favor of referencing it directly.